Jenkinsfile

```diff
@@ -30,6 +30,19 @@ def deleteDirWin() {
     deleteDir()
 }
 
+def sourceCodePaths(){
+    // These paths will be passed to git diff
+    // If there are changes to them, CI/CD will continue else skip
+    def paths = ['lib/stan_math', 'make', 'src/stan', 'src/test', 'Jenkinsfile', 'makefile', 'runTests.py']
```

Comment:
I think we would want these to ignore the docs file in math? Instead of defining all the folder paths that are not docs is there a way to exclude the docs folder from the path?

Reply:
Yes, it's so we can skip docs. 
The below commit `"lib/stan_math ':(exclude)lib/stan_math/doxygen"` translates to (example) `git diff 53835eef0a58dafeccaf1a31627075c4386572cd develop -- lib/stan_math ':(exclude)lib/stan_math/doxygen'` this should take care of excluding doxygen from diff.

Reply:
Nice!

Reply:
This looks fine, I am wondering if we just have it skip for everything we skip in math?

```
def paths = ['stan', 'make', 'lib', 'test', 'runTests.py', 'runChecks.py', 'makefile', 'Jenkinsfile', '.clang-format']
```
So these would be it:
```
def math_paths = ['lib/stan_math/stan', 'lib/stan_math/make', 'lib/stan_math/lib', 'lib/stan_math/test', 'lib/stan_math/runTests.py', 'lib/stan_math/runChecks.py', 'lib/stan_math/makefile', 'lib/stan_math/Jenkinsfile', 'lib/stan_math/.clang-format']
```

Reply:
Stan CI tests arent that long so this can be treated as optional.